lib/env_dpdk/env.c

```diff
@@ -31,17 +31,32 @@
  *   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "spdk/env.h"
+/*
+ *   Copyright (c) 1992-2017 NetApp, Inc.
+ *   All rights reserved.
+ */
+
 
+#include "spdk/env.h"
 #include <string.h>
 #include <unistd.h>
+#include <assert.h>
 
 #include <rte_config.h>
 #include <rte_cycles.h>
 #include <rte_malloc.h>
 #include <rte_mempool.h>
 #include <rte_memzone.h>
 #include <rte_version.h>
+#include <rte_lcore.h>
+
+/*
+ * Insure the definitions in spdk/env.h are equivelant to the DPDK definitions.
+ */
+
+static_assert(SPDK_MAX_LCORE == RTE_MAX_LCORE, "SPDK_MAX_LCORE != RTE_MAX_LCORE");
```

Comment:
These should use SPDK_STATIC_ASSERT() from spdk/assert.h for compatibility with old compilers that don't have static_assert.